tests/modules/ejector/test_ejector.py

```diff
@@ -0,0 +1,460 @@
+    ref_blockstamp: ReferenceBlockStamp,
+    chain_config: ChainConfig,
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    ejector.w3.cc.get_validators = Mock(
+        return_value=[
+            LidoValidatorFactory.build(),
+        ]
+        * 1024
+    )
+
+    ejector.get_chain_config = Mock(return_value=chain_config)
+
+    with monkeypatch.context() as m:
+        m.setattr(
+            ejector_module,
+            "is_partially_withdrawable_validator",
+            Mock(return_value=False),
+        )
+        m.setattr(
+            ejector_module,
+            "is_fully_withdrawable_validator",
+            Mock(return_value=False),
+        )
+
+        # no validators at all
+        result = ejector._get_sweep_delay_in_epochs(ref_blockstamp)
+        assert result == 0, "Unexpected sweep delay in epochs"
+
+    with monkeypatch.context() as m:
+        m.setattr(
+            ejector_module,
+            "is_partially_withdrawable_validator",
+            Mock(return_value=False),
+        )
+        m.setattr(
+            ejector_module,
+            "is_fully_withdrawable_validator",
+            Mock(return_value=True),
+        )
+
+        # all 1024 validators
+        result = ejector._get_sweep_delay_in_epochs(ref_blockstamp)
+        assert result == 1, "Unexpected sweep delay in epochs"
+
+
+@pytest.mark.unit
+@pytest.mark.usefixtures("contracts")
+def test_get_reserved_buffer(ejector: Ejector, blockstamp: BlockStamp) -> None:
+    result = ejector._get_reserved_buffer(blockstamp)
+    assert result == 31 * 10**18, "Unexpected reserved buffer"
+
+
+@pytest.mark.usefixtures("contracts")
+def test_get_total_balance(ejector: Ejector, blockstamp: BlockStamp) -> None:
+    ejector.w3.lido_contracts.get_withdrawal_balance = Mock(return_value=3)
+    ejector.w3.lido_contracts.get_el_vault_balance = Mock(return_value=17)
+    ejector._get_reserved_buffer = Mock(return_value=1)
+
+    result = ejector._get_total_balance(blockstamp)
+    assert result == 21, "Unexpected total balance"
+
+    ejector.w3.lido_contracts.get_withdrawal_balance.assert_called_once_with(blockstamp)
+    ejector.w3.lido_contracts.get_el_vault_balance.assert_called_once_with(blockstamp)
+    ejector._get_reserved_buffer.assert_called_once_with(blockstamp)
+
+
+class TestChurnLimit:
+    """_get_churn_limit tests"""
+
+    @pytest.fixture(autouse=True)
+    def mock_is_active_validator(self, monkeypatch: pytest.MonkeyPatch) -> Iterable:
+        with monkeypatch.context() as m:
+            m.setattr(
+                ejector_module,
+                "is_active_validator",
+                Mock(side_effect=lambda v, _: bool(v)),
+            )
+            yield
+
+    @pytest.mark.unit
+    @pytest.mark.usefixtures("consensus_client")
+    def test_get_churn_limit_no_validators(self, ejector: Ejector) -> None:
+        ejector.w3.cc.get_validators = Mock(return_value=[])
+        result = ejector._get_churn_limit(ref_blockstamp)
```

Comment:
`ref_blockstamp` is a fixture but you don't call it

Reply:
Because `ejector.w3.cc.get_validators = Mock(return_value=[1, 1, 0]) ` looks strange - the numbers are not a validators

Reply:
`ref_blockstamp` is passed as is, and test is not failed on attribute access since no validators are returned.

Reply:
But it looks like a test error. In this case, pass None or mock that causes an Exception if they called. Passing invalid object looks strange

Reply:
I described why it works, it is a mistake, apparently.